### PR TITLE
Upload bar

### PR DESCRIPTION
**Pull request by tenisakb**

### CODE REVIEW

File: src/redux/actions/rename-actions.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    type: RENAME_FILE_SUCCESS,
+    payload: { masterHandle, folder }
+  }),
+  renameFileError: ({ error }) => ({
+    type: RENAME_FILE_ERROR,
+    payload: { error }
+  })
+});
+
+export default ACTIONS;
```
Comment (EdmundMai) [inline src/redux/actions/rename-actions.ts]: it feels weird that renaming folders goes inside of the `folder` actions, but renaming files goes into a separate `rename` actions.

choose one:
1)  move all the "rename folder" stuff into this file
OR
2) rename this file to `file-actions`

up to you

File: src/redux/epics/rename-epic.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          });
+
+          return renameActions.renameFileSuccess({ masterHandle, folder });
+        }),
+        catchError(error => of(renameActions.renameFileError({ error })))
+      );
+    })
+  );
+
+export default combineEpics(renameFileEpic);
```
Comment (EdmundMai) [inline src/redux/epics/rename-epic.ts]: same comment about renaming. if you choose #1, move the rename folder stuff here. if you choose #2, rename this to `file-epic`.

File: src/components/file-manager/rename-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  height: 35px;
+  border: 0.5px solid;
+  font-size: 16px;
+  width: 80%;
+  margin: auto;
+  padding: 10px;
+`;
+
+const RenameModal = ({ close, isOpen, rename, name }) => {
+  const [newName, setNewName] = useState("");
```
[the remark this answers is not in the document]
Comment (tenisakb) [reply src/components/file-manager/rename-modal.tsx]: fixed

File: src/components/file-manager/rename-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  return (
+    <ThemeProvider theme={theme}>
+      <ModalProvider>
+        <StyledModal
+          isOpen={isOpen}
+          onBackgroundClick={() => [close(), setNewName("")]}
+          onEscapeKeydown={() => [close(), setNewName("")]}
```
[the remark this answers is not in the document]
Comment (tenisakb) [reply src/components/file-manager/rename-modal.tsx]: fixed

File: src/redux/actions/rename-actions.ts (hunk 1 above)
Comment (tenisakb) [reply src/redux/actions/rename-actions.ts]: fixed

File: src/redux/epics/rename-epic.ts (hunk 2 above)
Comment (tenisakb) [reply src/redux/epics/rename-epic.ts]: fixed

**Comment by EdmundMai** [comment]
Please double check your work:
- Two "Delete file" icons
- Rename folder / file icon is the same as the folder icon (confusing) and does not match the other icons

confusing

**Comment by funkydrummer** [comment]
I would use something like this for edit icon.

https://www.iconfinder.com/icons/2417839/edit_file_rename_icon

File: src/components/file-manager/dragDropContext.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+import { DragDropContext } from "react-dnd";
+import HTML5Backend from "react-dnd-html5-backend";
+
+export default DragDropContext(HTML5Backend);
```
Comment (EdmundMai) [inline src/components/file-manager/dragDropContext.ts]: 1) move this to `src/contexts`
2) rename it to match our other file names: `drag-drop-context.ts`

File: src/components/file-manager/file-manager-slide.tsx [hunk 6]
```diff
@@ -410,6 +420,151 @@ const FileManagerSlide = ({
...
+            ]}
+          >
+            <Tooltip content="Rename folder">
+              <TableIcon src={ICON_REMOVE} />
+            </Tooltip>
+          </ActionButton>
+        </Td>
+      </TrPointer>
+    );
+  };
```
Comment (EdmundMai) [inline src/components/file-manager/file-manager-slide.tsx]: move this into separate component file

File: src/components/file-manager/file-manager-slide.tsx [hunk 7]
```diff
@@ -410,6 +420,151 @@ const FileManagerSlide = ({
...
+            ]}
+          >
+            <Tooltip content="Rename file">
+              <TableIcon src={ICON_RENAME} />
+            </Tooltip>
+          </ActionButton>
+        </Td>
+      </Tr>
+    );
+  };
```
Comment (EdmundMai) [inline src/components/file-manager/file-manager-slide.tsx]: move this into separate component file

File: src/components/file-manager/rename-modal.tsx [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+  margin: auto;
+  padding: 10px;
+`;
+
+const RenameModal = ({ close, isOpen, rename, oldName }) => {
+  const [name, setName] = useState(oldName);
+
+  useEffect(() => {
+    setName(oldName);
+  }, [oldName]);
```
Comment (EdmundMai) [inline src/components/file-manager/rename-modal.tsx]: do you need this? shouldn't line 93 do the same thing?

File: src/config/index.ts [hunk 9]
```diff
@@ -332,3 +332,8 @@ export const DATA_TYPES_ICONS = [
     icon: ICON_PNG
   }
 ];
+
+export const DROP_TYPES = {
+  FILE: "FILE",
+  FOLDER: "FOLDER"
+};
```
Comment (EdmundMai) [inline src/config/index.ts]: use `enum`

File: src/components/file-manager/dragDropContext.ts (hunk 5 above)
Comment (tenisakb) [reply src/components/file-manager/dragDropContext.ts]: fixed

File: src/components/file-manager/file-manager-slide.tsx (hunk 6 above)
Comment (tenisakb) [reply src/components/file-manager/file-manager-slide.tsx]: fixed

File: src/components/file-manager/file-manager-slide.tsx (hunk 7 above)
Comment (tenisakb) [reply src/components/file-manager/file-manager-slide.tsx]: fixed

File: src/config/index.ts (hunk 9 above)
Comment (tenisakb) [reply src/config/index.ts]: fixed

File: src/components/file-manager/rename-modal.tsx (hunk 8 above)
Comment (tenisakb) [reply src/components/file-manager/rename-modal.tsx]: We need it.. it doesnt work without it

Comment (EdmundMai) [reply src/components/file-manager/rename-modal.tsx]: ok

**Comment by EdmundMai** [comment]
The code looks good but it does not work for me. Please QA

Screenshot_8_9_19__10_44_AM

**Comment by EdmundMai** [comment]
Everything looks like it works but I still see this compilation error. Do you not see it @tenisakb ?

```

ERROR in [at-loader] ./src/components/file-manager/index.tsx:87:7
    TS2339: Property 'createFolder' does not exist on type '(IntrinsicAttributes & IntrinsicClassAttributes<Component<Pick<{}, never> | (Pick<Pick<{}, never>...'.

```

**Comment by tenisakb** [comment]
> Everything looks like it works but I still see this compilation error. Do you not see it @tenisakb ?
> 
> ```
> 
> ERROR in [at-loader] ./src/components/file-manager/index.tsx:87:7
>     TS2339: Property 'createFolder' does not exist on type '(IntrinsicAttributes & IntrinsicClassAttributes<Component<Pick<{}, never> | (Pick<Pick<{}, never>...'.
> ```

Its in master as well.. I dont know why, but its not issue from this PR.

**Pull request merged**